### PR TITLE
Add nxos_lacp resource module

### PR DESCRIPTION
**Pull request by trishnaguha**
Signed-off-by: Trishna Guha <trishnaguha17@gmail.com>

##### SUMMARY

Add nxos_lacp resource module

##### ISSUE TYPE

- New Module Pull Request

##### COMPONENT NAME

modules/network/nxos/nxos_lacp.py

### CODE REVIEW

**Comment by NilashishC** [comment]
Looks like it needs a rebase.

File: lib/ansible/module_utils/network/nxos/argspec/lacp/lacp.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+
+class LacpArgs(object):
+    """The arg spec for the nxos_lacp module
+    """
+
+    def __init__(self, **kwargs):
+        pass
+
+    argument_spec = {'config': {'options': {'system': {'options': {'mac': {'options': {'address': {'type': 'str'},
```
Comment (danielmellado) [inline lib/ansible/module_utils/network/nxos/argspec/lacp/lacp.py]: Could we maybe sort this out so it's easier to see? I mean having something in the lines of
{'config':
    {'options':
        {'system':

And so. It'd be just so much more readable.

**Review by danielmellado — CHANGES_REQUESTED** [review]
Hi Trishna. overall IT LGTM, just some nit there and this needs a rebase as well. Thanks!

**Comment by ikhan2010** [comment]
@trishnaguha is it possible to get this merged this week?

File: lib/ansible/modules/network/nxos/nxos_lacp.py [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+    suboptions:
+      system:
+        description:
+          - LACP system options
+        type: dict
+        suboptions:
+          priority:
+            description:
+              - The system priority to use in LACP negotiations.
+            type: int
```
Comment (justjais) [inline lib/ansible/modules/network/nxos/nxos_lacp.py:60]: here `priority`/`mac` should be `required` param, correct?

**Review by danielmellado — APPROVED** [review]

**Comment by danielmellado** [comment]
Thanks for the changes @trishnaguha! Let's wait on CI and merge if everything's ok!

File: lib/ansible/modules/network/nxos/nxos_lacp.py (hunk 2 above)
Comment (trishnaguha) [reply lib/ansible/modules/network/nxos/nxos_lacp.py:60]: @justjais No there is no required parameter here. If no parameter is passed, the module won't do anything, it is not required or mandatory for nxos lacp. lacp can still be configured for interfaces if it's enabled even if these parameters are not configured.

**Pull request merged**